diff --git a/src/Make_mvc.mak b/src/Make_mvc.mak
index e1d42fdcb..31f15257a 100644
--- a/src/Make_mvc.mak
+++ b/src/Make_mvc.mak
@@ -1820,8 +1820,8 @@ $(PATHDEF_SRC): Make_mvc.mak
 	@echo char_u *default_vimruntime_dir = (char_u *)"$(VIMRUNTIMEDIR:\=\\)"; >> $(PATHDEF_SRC)
 	@echo char_u *all_cflags = (char_u *)"$(CC:\=\\) $(E_CFLAGS)"; >> $(PATHDEF_SRC)
 	@echo char_u *all_lflags = (char_u *)"$(LINK:\=\\) $(LINKARGS1:\=\\) $(E_LINKARGS2)"; >> $(PATHDEF_SRC)
-	@echo char_u *compiled_user = (char_u *)"$(USERNAME)"; >> $(PATHDEF_SRC)
-	@echo char_u *compiled_sys = (char_u *)"$(USERDOMAIN)"; >> $(PATHDEF_SRC)
+	@echo char_u *compiled_user = (char_u *)"Christian Brabandt <cb@256bit.org>"; >> $(PATHDEF_SRC)
+	@echo char_u *compiled_sys = (char_u *)""; >> $(PATHDEF_SRC)
 
 # End Custom Build
 proto.h: \
