diff --git a/src/if_ruby.c b/src/if_ruby.c
index 60ca11188..555b73043 100644
--- a/src/if_ruby.c
+++ b/src/if_ruby.c
@@ -604,9 +604,13 @@ rb_unexpected_type_stub(VALUE self, int t)
 }
 #  endif
 #  ifdef USE_TYPEDDATA
-void *rb_check_typeddata_stub(VALUE obj, const rb_data_type_t *data_type)
+void *
+rb_check_typeddata_stub(VALUE obj, const rb_data_type_t *data_type)
 {
-    return dll_rb_check_typeddata(obj, data_type);
+    void *result = dll_rb_check_typeddata(obj, data_type);
+    fprintf(stderr, "rb_check_typeddata_stub at %p called, result=%p\n", (void*)rb_check_typeddata_stub, result);
+    fflush(stderr);
+    return result;
 }
 #  endif
 # endif // ifndef PROTO
