diff --git a/nsis/gvim.nsi b/nsis/gvim.nsi
index 3fe0279b7..d846d1b7a 100644
--- a/nsis/gvim.nsi
+++ b/nsis/gvim.nsi
@@ -710,8 +710,19 @@ Function .onInit
   !insertmacro MUI_LANGDLL_DISPLAY
 !endif
 
-  # Check $VIM
-  ReadEnvStr $INSTDIR "VIM"
+	# If $VIM is set when executing the installer
+	# and no InstallDirectory has been given (not using D=C:\Vim),
+	# use $VIM as Install directory,
+  # Note: this cleans $INSTDIR if $VIM is not set!
+	!ifdef WIN64
+		${If} $INSTDIR == "$PROGRAMFILES64\Vim"
+			ReadEnvStr $INSTDIR "VIM"
+		${EndIf}
+	!else
+		${If} $INSTDIR == "$PROGRAMFILES\Vim"
+			ReadEnvStr $INSTDIR "VIM"
+		${EndIf}
+	!endif
 
   call CheckOldVim
   Pop $3
@@ -725,6 +736,9 @@ Function .onInit
       StrCpy $INSTDIR $3
     ${EndIf}
   ${EndIf}
+ MessageBox MB_YESNO "InstDir: $INSTDIR. Continue?" IDYES NoAbort
+      Abort ; causes installer to quit.
+  NoAbort:
 
   # If did not find a path: use the default dir.
   ${If} $INSTDIR == ""
